scipy/_lib/_array_api.py

```diff
@@ -381,6 +381,24 @@ def xp_assert_less(actual, desired, check_namespace=True, check_dtype=True,
                                         err_msg=err_msg, verbose=verbose)
 
 
+def assert_array_almost_equal(actual, desired, decimal=6, *args, **kwds):
+    """Backwards compatible replacement. In new code, use xp_assert_close instead.
+    """
+    rtol, atol = 0, 1.5*10**(-decimal)
+    return xp_assert_close(actual, desired,
+                           atol=atol, rtol=rtol, check_dtype=False, allow_0d=True,
+                           *args, **kwds)
+
+
+def assert_almost_equal(actual, desired, decimal=7, *args, **kwds):
+    """Backwards compatible replacement. In new code, use xp_assert_close instead.
+    """
+    rtol, atol = 0, 1.5*10**(-decimal)
+    return xp_assert_close(actual, desired,
+                            atol=atol, rtol=rtol, check_dtype=False, allow_0d=True,
+                            *args, **kwds)
```

Comment:
let's make these typed since the rest of the file is. Also indentation on lines 398, 399

Reply:
`xp_assert_*` functions are not: https://github.com/scipy/scipy/blob/main/scipy/_lib/_array_api.py#L312

Reply:
Apologies, you're right (https://github.com/scipy/scipy/pull/20621#issue-2273073907)